src/main/scala/edu/ie3/simona/agent/grid/ReceivedValuesStore.scala

```diff
@@ -47,7 +47,7 @@ final case class ReceivedValuesStore private (
 object ReceivedValuesStore {
 
   type NodeToReceivedPower =
-    Map[UUID, Vector[(ActorRef, Option[PowerResponseMessage])]]
+    Map[UUID, Map[ActorRef, Option[PowerResponseMessage]]]
```

Comment:
Since we changed to use a map now doesn't that mean the whole boolean replace part does no longer work? If so is that safe? Also if so we might want to remove the functionality.

Reply:
Well the `replace` boolean is now only used in `getNodeUuidForSender()` (which was `uuid()` before), and it only functions as some sort of safeguard, where an exception is thrown if your expectation is not met. Meaning the if you expect to not have received data before but you actually *did*, the uuid wouldn't be returned and you would throw in `updateNodalReceivedPower`. 

So yes, I think `replace` doesn't serve a functionality that is really required here, and I think it's up to personal preference whether one would want to have it removed.

Reply:
Okay I get it now - thanks for clarification 🙂